src/s_nan.c

```diff
@@ -36,7 +36,7 @@
 
 #include "math_private.h"
 
-#if !defined(__APPLE__) && !defined(__FreeBSD__)
+#if !defined(__APPLE__) && !defined(__FreeBSD__) && !defined(__DragonFly__)
```

Comment:
Should this maybe just be a check on Windows and Linux? Presumably all BSDs should be excluded here.

Reply:
Isn't there a common flag to detect any kind of BSD? Here the test is clearly specific to a family of OSes.

Reply:
Interestingly, this builds fine on OpenBSD without changing this line, and adding `!defined(__OpenBSD__)` actually causes the tests to fail with an undefined reference to `digittoint`. So I guess this is better left as an explicit list.